src/main/java/edu/ucsb/cs156/organic/controllers/CoursesController.java

```diff
@@ -132,4 +132,55 @@ public Iterable<Staff> getStaff(
         return courseStaff;
     }
 
+    @Operation(summary = "Update information for a course")
+    // allow for roles of ADMIN or INSTRUCTOR but only if the user is a staff member
+    // for the course
+    @PreAuthorize("hasAnyRole('ROLE_ADMIN', 'ROLE_INSTRUCTOR')")
+    @PutMapping("/update")
+    public Course updateCourse(
+            @Parameter(name = "id") @RequestParam Long id,
+            @Parameter(name = "name", description = "course name, e.g. CMPSC 156") @RequestParam String name,
+            @Parameter(name = "school", description = "school abbreviation e.g. UCSB") @RequestParam String school,
+            @Parameter(name = "term", description = "quarter or semester, e.g. F23") @RequestParam String term,
+            @Parameter(name = "start", description = "in iso format, i.e. YYYY-mm-ddTHH:MM:SS; e.g. 2023-10-01T00:00:00 see https://en.wikipedia.org/wiki/ISO_8601") @RequestParam("start") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime start,
+            @Parameter(name = "end", description = "in iso format, i.e. YYYY-mm-ddTHH:MM:SS; e.g. 2023-12-31T11:59:59 see https://en.wikipedia.org/wiki/ISO_8601") @RequestParam("end") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime end,
+            @Parameter(name = "githubOrg", description = "for example ucsb-cs156-f23") @RequestParam String githubOrg)
+            throws JsonProcessingException {
+
+        Course course = courseRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(Course.class, id.toString()));
+
+        // Check if the current user is a staff member for this course. If not, throw
+        // AccessDeniedException
+
+        Iterable<Staff> courseStaff = courseStaffRepository.findByCourseId(course.getId());
+        User u = getCurrentUser().getUser();
+        boolean isStaffOrAdmin = u.isAdmin();
+        for (Staff staff : courseStaff) {
+            if (staff.getGithubId().equals(u.getGithubId())) {
+                isStaffOrAdmin = true;
+                break;
+            }
+        }
+
+        if (!isStaffOrAdmin) {
+            throw new AccessDeniedException(
+                    "User is not a staff member for this course");
+        }
```

Comment:
Please see: https://ucsb-cs156.github.io/topics/sql/sql_queries.html

That article was made exactly for this situation.

There's a better approach here that replaces 14 lines of code with if tests and for loops with five lines of straight line code.